log4j-api/src/main/java/org/apache/logging/log4j/status/StatusConsoleListener.java

```diff
@@ -56,50 +54,59 @@ public StatusConsoleListener(final Level level) {
      * @throws NullPointerException on null {@code level} or {@code stream}
      */
     public StatusConsoleListener(final Level level, final PrintStream stream) {
-        this(level, stream, SimpleLoggerFactory.getInstance());
-    }
-
-    StatusConsoleListener(final Level level, final PrintStream stream, final SimpleLoggerFactory loggerFactory) {
-        this.level = Objects.requireNonNull(level, "level");
-        this.stream = Objects.requireNonNull(stream, "stream");
-        this.logger = Objects.requireNonNull(loggerFactory, "loggerFactory")
-                .createSimpleLogger(
-                        "StatusConsoleListener", level, ParameterizedNoReferenceMessageFactory.INSTANCE, stream);
+        this.level = requireNonNull(level, "level");
+        this.stream = requireNonNull(stream, "stream");
     }
 
     /**
      * Sets the level to a new value.
-     * @param level The new Level.
+     *
+     * @param level the new level
+     * @throws NullPointerException on null {@code level}
      */
     public void setLevel(final Level level) {
-        this.level = level;
+        this.level = requireNonNull(level, "level");
     }
 
     /**
-     * Return the Log Level for which the Listener should receive events.
-     * @return the Log Level.
+     * Sets the output stream to a new value.
+     *
+     * @param stream the new output stream
+     * @throws NullPointerException on null {@code stream}
+     * @since 2.23.0
+     */
+    public void setStream(final PrintStream stream) {
+        this.stream = requireNonNull(stream, "stream");
+    }
+
+    /**
+     * Returns the level for which the listener should receive events.
+     *
+     * @return the log level
      */
     @Override
     public Level getStatusLevel() {
-        return this.level;
+        return level;
     }
 
     /**
      * Writes status messages to the console.
-     * @param data The StatusData.
+     *
+     * @param data a status data
+     * @throws NullPointerException on null {@code data}
      */
     @Override
     public void log(final StatusData data) {
-        logger
-                // Logging using _only_ the following 4 fields set by `StatusLogger#logMessage()`:
-                .atLevel(data.getLevel())
-                .withThrowable(data.getThrowable())
-                .withLocation(data.getStackTraceElement())
-                .log(data.getMessage());
+        requireNonNull(data, "data");
+        if (data.getLevel().isMoreSpecificThan(level)) {
+            final String formattedStatus = data.getFormattedStatus();
+            stream.println(formattedStatus);
```

Comment:
Call me opinionated, but I (strongly?) prefer `var a = a(); var b = b(a); c(b)` over `c(b(a))`, since
1. Variables give you documentation opportunity (e.g., `boolean validUserName = userName.matches(userNamePattern)`)
2. One-statement-per-line is easier to read and understand
3. Variables allow you to track intermediate values while stepping through code in a debugger. AFAIK, this is not possible in `c(b(a))` form.